src/ore/src/now.rs

```diff
@@ -9,35 +9,74 @@
 
 //! Now utilities.
 
+use std::fmt;
+use std::ops::Deref;
+use std::sync::Arc;
 use std::time::SystemTime;
 
+use lazy_static::lazy_static;
+
 #[cfg(feature = "chrono")]
 use chrono::{DateTime, TimeZone, Utc};
 
 /// A type representing the number of milliseconds since the Unix epoch.
 pub type EpochMillis = u64;
 
+/// Converts epoch milliseconds to a DateTime.
+#[cfg(feature = "chrono")]
+pub fn to_datetime(millis: EpochMillis) -> DateTime<Utc> {
+    let dur = std::time::Duration::from_millis(millis);
+    Utc.timestamp(dur.as_secs() as i64, dur.subsec_nanos())
+}
+
 /// A function that returns system or mocked time.
-pub type NowFn = fn() -> EpochMillis;
+// This is a newtype so that it can implement `Debug`, as closures don't
+// implement `Debug` by default. It derefs to a callable so that it is
+// ergonomically equivalent to a closure.
+#[derive(Clone)]
+pub struct NowFn(Arc<dyn Fn() -> EpochMillis + Send + Sync>);
```

Comment:
Maybe this is clear to someone with more rust experience but: does the use of `dyn` in the `Deref` trait impose an additional lookup that wasn't present in the previous version? Or is there something smart enough to compile away the dyn lookup?

Reply:
I don't actually have a good model for what `dyn Fn` compiles down to. 🤷🏽‍♂️